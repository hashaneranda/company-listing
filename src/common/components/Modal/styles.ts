import styled from 'styled-components';

import { Modal, Paper } from '@material-ui/core';

export const ModalComp = styled(Modal)`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const PaperWrapper = styled(Paper)`
  position: relative;
  padding-top: 3em;
  background: ${props => props.theme.palette.background.secondary};
  color: ${props => props.theme.palette.typography.main};

  &:focus {
    outline: none;
  }
`;

export const Image = styled.img.attrs(props => ({
  src: props.src,
}))`
  position: absolute;
  right: 32px;
  top: 28px;

  width: 15px;
  height: 15px;

  &:hover {
    cursor: pointer;
    transform: scale(1.05);
  }
`;
