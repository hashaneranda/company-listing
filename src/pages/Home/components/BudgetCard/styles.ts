import styled from 'styled-components';

export const Container = styled.div`
  padding: 1em 2em;
  display: grid;
  grid-template-columns: minmax(250px, 25%) 1fr;
`;

export const LeftContainer = styled.div`
  padding: 1em 2em;
  display: grid;
  place-items: center;
`;

export const RightContainer = styled.div`
  padding: 1em 2em;
  display: flex;
  flex-direction: column;
`;

export const BudgetRow = styled.div`
  display: flex;
  flex-direction: row;
`;
