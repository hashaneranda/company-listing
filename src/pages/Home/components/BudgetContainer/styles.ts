import styled from 'styled-components';

export const Card = styled.div`
  padding: 1em 2em;
  display: grid;
  grid-template-columns: minmax(50px, 25%) 1fr;

  &.primary {
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 3em;

    h4 {
      font-size: 28px;
      color: ${props => props.theme.palette.color.primary};
      margin-top: 0.2em;
    }

    img {
      width: 55px;
    }
  }
`;

export const CardWrapper = styled.div`
  display: grid;
  place-items: center;

  img {
    width: 35px;
  }
`;

export const CardBody = styled.div`
  display: flex;
  flex-direction: column;

  color: ${props => props.theme.palette.typography.secondary};

  p {
    color: ${props => props.theme.palette.typography.secondary};
  }

  h4 {
    color: ${props => props.theme.palette.typography.main};
  }
`;
