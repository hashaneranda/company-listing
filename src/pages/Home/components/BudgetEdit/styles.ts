import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 40em;
  padding: 1em 2em;

  line-height: 24px;
  overflow: hidden;
  position: relative;

  background: ${props => props.theme.palette.background.secondary};

  border-radius: 12px;
  box-shadow: 0 0 20px ${props => props.theme.palette.shadow.main}, 0 0px 40px ${props => props.theme.palette.shadow.secondary};
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  padding: 2em 0;

  .budget_edit__btn {
    height: 1em;
    margin-top: 1em;
  }
`;
