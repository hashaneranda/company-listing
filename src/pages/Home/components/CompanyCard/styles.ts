import styled from 'styled-components';
import tinycolor from 'tinycolor2';

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 40em;
  padding: 1em 2em;

  line-height: 24px;
  overflow: hidden;
  position: relative;

  border-radius: 12px;
  box-shadow: 0 0 20px ${props => props.theme.palette.shadow.main}, 0 0px 40px ${props => props.theme.palette.shadow.secondary};

  ${props => props.theme.title === 'dark' && `background: ${props.theme.palette.background.secondary}`};

  &:hover {
    cursor: pointer;
    background: ${props => props.theme.palette.background.secondary};

    ${props =>
      props.theme.title === 'dark' && `background: ${tinycolor(props.theme.palette.background.secondary).lighten(5).toHexString()}`};
  }
`;

export const CardHeader = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  p {
    color: ${props => props.theme.palette.typography.secondary};
  }

  .date_container {
    display: flex;
    flex-direction: column;
    text-align: right;
  }
`;

export const CardBody = styled.div`
  display: grid;
  grid-template-rows: auto 1fr auto;
  text-align: center;
`;
