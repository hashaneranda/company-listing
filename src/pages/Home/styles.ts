import styled from 'styled-components';

export const Container = styled.div`
  padding: 1em 2em;
  display: grid;
  place-items: center;
  min-height: 100vh;
`;

export const MainWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
`;

export const CompanyWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 2em;

  margin-top: 3em;
`;
