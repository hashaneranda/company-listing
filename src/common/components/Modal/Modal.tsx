import React from 'react';

import { ModalProps } from '@material-ui/core/Modal';
import { Grow } from '@material-ui/core';

//assets
import { closeBtn } from 'config/images';

//styles
import { ModalComp, PaperWrapper, Image } from './styles';

interface ModalComponentProps extends ModalProps {
  open: boolean;
  // children: React.ReactNode | React.ReactNode[],
  handleClose: () => any;
  showCloseBtn?: boolean;
}

/**
 *Modal Component
 *
 * @param {*} {
 *   children - modal body,
 *   open = false - modal open status,
 *   handleClose = () => {} - function to close modal,
 *   showCloseBtn = false - show default close button,
 *   ...props
 * }
 * @returns Modal Component
 */
const ModalComponent = ({ children, open = false, handleClose = () => null, showCloseBtn = false, ...props }: ModalComponentProps) => {
  return (
    <ModalComp
      aria-labelledby='simple-modal-title'
      aria-describedby='simple-modal-description'
      open={open}
      onClose={handleClose}
      {...props}
    >
      <Grow in={open}>
        <PaperWrapper>
          {showCloseBtn ? <Image src={closeBtn.default} alt='close' onClick={handleClose} data-testid='modal_close' /> : ''}
          {children}
        </PaperWrapper>
      </Grow>
    </ModalComp>
  );
};

export default ModalComponent;
