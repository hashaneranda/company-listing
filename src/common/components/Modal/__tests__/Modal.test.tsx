import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import Modal from '../Modal';

import { StylesProvider } from '@material-ui/core/styles';
import Theme from 'theme/theme';
import ThemeSwitcher from 'theme/themeSwitcher';

const renderWithStyles = (component: React.ReactNode) => {
  return render(
    <StylesProvider injectFirst>
      <ThemeSwitcher>
        <Theme>{component}</Theme>
      </ThemeSwitcher>
    </StylesProvider>,
  );
};

afterEach(cleanup);

describe('Modal Componenet', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderWithStyles(
      <Modal>
        <h1>Test</h1>
      </Modal>,
    );

    expect(
      asFragment(
        <Modal>
          <h1>Test</h1>
        </Modal>,
      ),
    ).toMatchSnapshot();
  });

  it('children renders correctly', () => {
    const { getByText } = renderWithStyles(
      <Modal open>
        <h1>Test</h1>
      </Modal>,
    );
    expect(getByText('Test')).toBeInTheDocument();
  });

  it('close button works correctly', () => {
    let testToggler = true;
    const closeActionMock = () => {
      testToggler = false;
    };

    const { getByTestId } = renderWithStyles(
      <Modal handleClose={() => closeActionMock()} showCloseBtn open>
        Test
      </Modal>,
    );
    fireEvent.click(getByTestId('modal_close'));

    expect(testToggler).toEqual(false);
  });
});
