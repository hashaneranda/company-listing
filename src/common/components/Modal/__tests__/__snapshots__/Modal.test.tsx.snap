// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Componenet should take a snapshot 1`] = `<DocumentFragment />`;
